### ISSUE TITLE
Editor's code collapse does not work with upper code already collapsed

### ISSUE THREAD

**Issue by michelkaporin** [Microsoft/vscode#25127]
- VSCode Version: 1.12
- OS Version: Win 10

Steps to Reproduce:

1. Create new XML file with the following content:
```
<?xml version= "1.0" encoding= "utf-8" ?>
<root>
	<test>
	<nested>
		Hello world!
	</nested>
	<nested>
		Hello world!
	</nested>
	</test>
</root>
```
2. Collapse first `<nested/>` node
3. Try collapsing second `<nested/>` node.
-> Fails to collapse it.

![expanding](https://cloud.githubusercontent.com/assets/2239563/25267120/eef2727e-2673-11e7-9b7f-2c03dced0887.gif)

**Comment by alexandrudima** [comment]
Fixed via 3860b1b79a56facd94627ef19be096cf4afbcc2a

**Comment by michelkaporin** [comment]
Thank you @alexandrudima!
